src/Table.js

```diff
@@ -2,63 +2,88 @@ import classNames from 'classnames';
+import { createBootstrapComponent } from './ThemeProvider';
 
-const propTypes = {
-  inverse: PropTypes.bool,
-  striped: PropTypes.bool,
-  bordered: PropTypes.bool,
-  condensed: PropTypes.bool,
-  hover: PropTypes.bool,
-  responsive: PropTypes.bool
-};
+class Table extends React.Component {
+  static propTypes = {
+    /**
+     * @default 'table'
+     */
+    bsPrefix: PropTypes.string,
 
-const defaultProps = {
-  inverse: false,
-  bordered: false,
-  condensed: false,
-  hover: false,
-  responsive: false,
-  striped: false
-};
+    /**
+     * Adds zebra-striping to any table row within the `<tbody>`.
+     */
+    striped: PropTypes.bool,
+
+    /**
+     * Adds borders on all sides of the table and cells.
+     */
+    bordered: PropTypes.bool,
+
+    /**
+     * Enable a hover state on table rows within a `<tbody>`.
+     */
+    hover: PropTypes.bool,
+
+    /**
+     * Make tables more compact by cutting cell padding in half by setting
+     * size as `sm`.
+     */
+    size: PropTypes.string,
+
+    /**
+     * Invert the colors of the table — with light text on dark backgrounds
+     * by setting variant as `dark`.
+     */
+    variant: PropTypes.string,
+
+    /**
+     * Responsive tables allow tables to be scrolled horizontally with ease.
+     * Across every breakpoint, use `responsive` for horizontally
+     * scrolling tables. Responsive tables are wrapped automatically in a `div`.
+     * Use `responsive="sm"`, `responsive="md"`, `responsive="lg"`, or
+     * `responsive="xl"` as needed to create responsive tables up to
+     * a particular breakpoint. From that breakpoint and up, the table will
+     * behave normally and not scroll horizontally.
+     */
+    responsive: PropTypes.oneOfType([PropTypes.bool, PropTypes.string])
+  };
 
-class Table extends React.Component {
   render() {
     const {
-      inverse,
+      bsPrefix,
       striped,
       bordered,
-      condensed,
       hover,
+      size,
+      variant,
       responsive,
-      className,
       ...props
     } = this.props;
 
-    const [bsProps, elementProps] = splitBsProps(props);
+    const classes = classNames(
+      bsPrefix,
+      variant && `${bsPrefix}-${variant}`,
+      size && `${bsPrefix}-${size}`,
+      striped && `${bsPrefix}-striped`,
+      bordered && `${bsPrefix}-bordered`,
+      hover && `${bsPrefix}-hover`
+    );
 
-    const classes = {
-      ...getClassSet(bsProps),
-      [prefix(bsProps, 'responsive')]: responsive,
-      [prefix(bsProps, 'inverse')]: inverse,
-      [prefix(bsProps, 'striped')]: striped,
-      [prefix(bsProps, 'bordered')]: bordered,
-      [prefix(bsProps, 'condensed')]: condensed,
-      [prefix(bsProps, 'hover')]: hover
-    };
+    const table = <table {...props} className={classes} />;
 
-    return (
-      <table {...elementProps} className={classNames(className, classes)} />
-    );
+    if (responsive) {
+      let reponsiveClass = `${bsPrefix}-responsive`;
```

Comment:
"responsiveClass"

Reply:
Good catch, thanks! :) I have changed and pushed.